executors/stable/executor.py

```diff
@@ -197,152 +82,89 @@ def _h_and_w_from_parameters(self, parameters, opt):
-                                        mime_type='image/png',
-                                        tags={
-                                            'text': prompt,
-                                            'generator': 'stable-diffusion',
-                                            'request_time': request_time,
-                                            'created_time': time.time(),
-                                        },
-                                    ).convert_blob_to_datauri()
-                                    _d.text = prompt
-                                    d.matches.append(_d)
+
+        for d in docs:
+            batch_size = n_samples
+            prompt = d.text
+            assert prompt is not None
+
+            self.logger.info(f'stable diffusion start {num_images} images, prompt "{prompt}"...')
+            for i in trange(n_iter, desc="Sampling"):
+                samples, extra_data = self.inference_engine.sample(
+                    prompt,
+                    batch_size,
+                    sampler,
+                    seed + i,
+                    steps,
+                    height=height,
+                    scale=scale,
+                    width=width,
+                )
+
+                (
+                    conditioning,
+                    images,
+                ) = itemgetter('conditioning', 'images')(extra_data)
+
+                for img in images:
+                    buffered = BytesIO()
+                    img.save(buffered, format='PNG')
+
+                    samples_buffer = BytesIO()
+                    torch.save(samples, samples_buffer)
+                    samples_buffer.seek(0)
+
+                    _d = Document(
+                        embedding=conditioning,
+                        blob=buffered.getvalue(),
+                        mime_type='image/png',
+                        tags={
+                            'latent_repr': base64.b64encode(
+                                samples_buffer.getvalue()).decode(),
+                            'request': {
+                                'api': 'txt2img',
+                                'height': height,
+                                'num_images': num_images,
+                                'sampler': sampler,
+                                'scale': scale,
+                                'seed': seed,
+                                'steps': steps,
+                                'width': width,
+                            },
+                            'text': prompt,
+                            'generator': 'stable-diffusion',
+                            'request_time': request_time,
+                            'created_time': time.time(),
+                        },
+                    ).convert_blob_to_datauri()
+                    _d.text = prompt
+                    d.matches.append(_d)
+
+                torch.cuda.empty_cache()
```

Comment:
Why is this one here ? Would not ot drop the model weight as well ?

Reply:
`torch.cuda.empty_cache()` will just clear out all the dangling references to GPU memory that are no longer in use from the `sample` function, without effecting the model at all. I find it helps keep memory down. The model itself always stays in memory because it has active reference in the `StableDiffusionInference` class.

Reply:
Hmm okay fair enough